async-http-client/src/main/scala/org/http4s/client/asynchttpclient/AsyncHttpClient.scala

```diff
@@ -0,0 +1,134 @@
+package org.http4s
+package client
+package asynchttpclient
+
+import org.asynchttpclient.AsyncHandler.State
+import org.asynchttpclient.request.body.generator.{InputStreamBodyGenerator, BodyGenerator}
+import org.asynchttpclient.{Request => AsyncRequest, Response => _, _}
+import org.asynchttpclient.handler.StreamedAsyncHandler
+
+import org.http4s.util.task
+import org.reactivestreams.{Subscription, Subscriber, Publisher}
+import scodec.bits.ByteVector
+
+import scala.collection.JavaConverters._
+
+import scalaz.stream.io._
+import scalaz.stream.async
+import scalaz.concurrent.Task
+
+import scala.concurrent.Promise
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import org.log4s.getLogger
+
+object AsyncHttpClient {
+  private[this] val log = getLogger
+
+  val defaultConfig = new DefaultAsyncHttpClientConfig.Builder()
```

Comment:
Do we want the defaults for all the clients to be the same? If so, maybe we should move information such as max connections etc to the client package.